client/lib/simple-payments/utils.js

```diff
@@ -5,3 +5,8 @@ import { SIMPLE_PAYMENTS_PRODUCT_POST_TYPE } from 'lib/simple-payments/constants
 
 export const isValidSimplePaymentsProduct = product =>
 	product.type === SIMPLE_PAYMENTS_PRODUCT_POST_TYPE && product.status === 'publish';
+
+export const decodeProductAttribute = productAttribute =>
+	productAttribute.replace( /(?:&#(\d+);)|(&amp;)/g, ( match, charCode, amp ) =>
```

Comment:
I looked in wp-calypso to see if we had any other character decoding tools but I couldn't find anything. I'd gladly replace this if anyone knows of an existing lib or helper. 

But this a JSX issue : https://shripadk.github.io/react/docs/jsx-gotchas.html#html-entities
> The easiest [option] is to write unicode character directly in Javascript.
That's what's happening here. Converting something like `&#8220;` into `"`. 
I'll write a comment near the regexp to help clear up why someone is wiking  ;) 

`String.fromCharCode` then returns the unicode value which JSX will render correctly